babel/localedata.py

```diff
@@ -54,6 +56,8 @@ def exists(name):
 
     :param name: the locale identifier string
     """
+    if not name or not isinstance(name, string_types):
```

Comment:
Same here: It would probably be better and/or more Pythonic to just cast to the text type here, instead of strictly requiring a string type.

Reply:
Perhaps that's functionality that could be implemented at a later date?

Using `isinstance()` would make it match the currency functions, e.g., https://github.com/python-babel/babel/blob/master/babel/numbers.py#L82.

I'm also not sure whether accepting an object for which `str(obj)` returns a locale should be supported long-term?

Reply:
I suppose you're correct. Didn't remember the currency functions already acted like this anyway.